WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Welcome/MigrationWelcomeViewController.swift

```diff
@@ -10,14 +10,18 @@ final class MigrationWelcomeViewController: UIViewController {
 
     private let tableView = UITableView(frame: .zero, style: .plain)
 
-    private let headerView: MigrationHeaderView = {
-        let view = MigrationHeaderView()
+    private lazy var headerView: MigrationHeaderView = {
+        let view = MigrationHeaderView(configuration: viewModel.configuration.headerConfiguration)
         view.translatesAutoresizingMaskIntoConstraints = true
         view.directionalLayoutMargins = Constants.tableHeaderViewMargins
         return view
     }()
 
-    private let bottomSheet = MigrationActionsView()
+    private lazy var bottomSheet: MigrationActionsView = {
+        let actionsView = MigrationActionsView(configuration: viewModel.configuration.actionsConfiguration)
```

Comment:
Regarding the `MigrationActionsView` class, I just want to mention that I was planning to rename it to something like `ActionsSheetView`, so that the component can be re-used in other screens. 

We already have a similar component in the Jetpack / WordPress Prologue Screen, and it's very possible that we might see this component again in another screen.

But right now, this component is more dependent on the `Migration` feature because the `MigrationActionsViewConfiguration` type is dependent on `MigrationStep` object.

It would be great to have an initializer that doesn't depend on `MigrationStep` and another "convenience" init that can depend on it. In other words, it would be great to have the 2 initializers:

<img width="400" alt="CleanShot 2022-11-07 at 16 37 56@2x" src="https://user-images.githubusercontent.com/9609223/200351159-168e2a6d-d8ab-49bd-91e5-4a32f3881dbf.png">

We can achieve the above by moving the `init(step: MigrationStep...)` init to an extension:

```swift
// This struct can be defined in the same file as `MigrationActionsView.swift` file or even embedded in the `MigrationActionsView` class.
struct MigrationActionsViewConfiguration {
    let primaryTitle: String?
    let secondaryTitle: String?
    let primaryHandler: (() -> Void)?
    let secondaryHandler: (() -> Void)?
}

// Have this extension in another file.
extension MigrationActionsViewConfiguration {

    init(step: MigrationStep, primaryHandler: (() -> Void)? = nil, secondaryHandler: (() -> Void)? = nil) {
        self.primaryHandler = primaryHandler
        self.secondaryHandler = secondaryHandler
        self.primaryTitle = Appearance.primaryTitle(for: step)
        self.secondaryTitle = Appearance.secondaryTitle(for: step)
    }

    private enum Appearance {
        // ...
    }
}
```

Would look like to hear your thoughts on making this component reusable in other screens other than `Migration` screens.

_P.S: This is definitely not urgent and we can address it later_

Reply:
This makes total sense. I'd think that, if we want to make this view reusable, we should then move it to `WordPressUI`. What do you say we do this refactor at the same time we move it there?

Reply:
@Gio2018 Sounds good to me 👍